Sources/MoEngage-Swift-Segment/MoEngageDestination.swift

```diff
@@ -0,0 +1,217 @@
+            }
+            
+            if let gender = (traits[UserAttributes.gender.rawValue] as? String)?.lowercased() {
+                if gender == "m" || gender == "male" {
+                    MoEngageSDKAnalytics.sharedInstance.setGender(.male, forAppID: moengageSettings?.apiKey)
+                }
+                else if gender == "f" || gender == "female" {
+                    MoEngageSDKAnalytics.sharedInstance.setGender(.female, forAppID: moengageSettings?.apiKey)
+                }
+                else {
+                    MoEngageSDKAnalytics.sharedInstance.setGender(.others, forAppID: moengageSettings?.apiKey)
+                }
+            }
+                        
+            if let phone = traits[UserAttributes.phone.rawValue] as? String {
+                MoEngageSDKAnalytics.sharedInstance.setMobileNumber(phone, forAppID: moengageSettings?.apiKey)
+            }
+       
+            if let isoDate = traits[UserAttributes.isoDate.rawValue] as? [String: Any] {
+                if let date = isoDate["date"] as? Date, let attributeName = isoDate["attributeName"] as? String {
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttributeDate(date, withAttributeName: attributeName, forAppID: moengageSettings?.apiKey)
+                }
+            }
+            
+            if let location = traits[UserAttributes.location.rawValue] as? [String: Any] {
+                if let latitute = location["latitude"] as? Double, let longitude = location["longitude"] as? Double {
+                    MoEngageSDKAnalytics.sharedInstance.setLocation(MoEngageGeoLocation(withLatitude: latitute, andLongitude: longitude))
+                }
+            }
+            
+            let moengageTraits = UserAttributes.allCases.map { $0.rawValue }
+            
+            for trait in traits where !moengageTraits.contains(trait.key) {
+                switch trait.value {
+                case let val as Date:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttributeDate(val, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                default:
+                    MoEngageSDKAnalytics.sharedInstance.setUserAttribute(trait.value, withAttributeName: trait.key, forAppID: moengageSettings?.apiKey)
+                }
+            }
+        }
+
+        return event
+    }
+
+    public func track(event: TrackEvent) -> TrackEvent? {
+        if var generalAttributeDict = event.properties?.dictionaryValue {
+            var dateAttributeDict: [String : Date] = [:]
+            
+            for key in generalAttributeDict.keys {
+                let val = generalAttributeDict[key]
+                if val is String {
+                    let converted_date = date(fromISOdateStr: val as? String)
+                    if let converted_date {
+                        dateAttributeDict[key] = converted_date
+                        generalAttributeDict.removeValue(forKey: key)
```

Comment:
@RakshithaAcharya It would crash if I am trying to access the value that was not present or if I was looping through using the index. I didnt see any crash here. It just shifts to the next element after deleting the item.